### PR TITLE
raster-symbology-editor

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Raster-Symbology-Editor anpassen](https://trello.com/c/eO5c6kW2/39-raster-symbology-editor-anpassen)

File: projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            }
+            this.propagateChange();
+        }
+    }
+
+    updateColor(color: string): void {
+        // TODO: should this really clone?
+        if (color) {
+            const clr = Color.fromRgbaLike(stringToRgbaStruct(color));
+            if (!clr.equals(this._colorBreakpoint.rgba)) {
```
Comment (jdroenner) [inline projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts:63]: there is no rgba in ColorBreakpoint

File: projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            this._colorBreakpoint = breakpoint;
+        }
+    }
+
+    updateValue(value: number | string): void {
+        if (value && value !== this.colorBreakpoint.value) {
+            if (this.inputType === 'number' && typeof value === 'string') {
+                this.colorBreakpoint.setValue(parseFloat(value as string));
+            } else {
+                this.colorBreakpoint.setValue(value);
```
Comment (jdroenner) [inline projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts:53]: can we check that this is a number?

File: projects/wave-core-new/src/lib/colors/color-map-selector/color-map-selector.component.ts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        const colorMap: Array<RgbaTuple> = this.form.controls['colorMap'].value;
+        const colorMapSteps: number = this.form.controls['colorMapSteps'].value;
+        const colorMapReverseColors: boolean = this.form.controls['colorMapReverseColors'].value;
+        const bounds: {min: number; max: number} = this.form.controls['bounds'].value;
+
+        // TODO: allow different gradient types
+
+        const breakpoints = new Array<ColorBreakpoint>();
+
+        const stepSize = 1 / (colorMapSteps - 1);
```
Comment (jdroenner) [inline projects/wave-core-new/src/lib/colors/color-map-selector/color-map-selector.component.ts:211]: we coould add a test for this logic ;)

**Review by jdroenner — COMMENTED** [review]
great job. If possible address the comments before merging.

File: projects/wave-core-new/src/lib/colors/color-map-selector/color-map-selector.component.ts (hunk 3 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/colors/color-map-selector/color-map-selector.component.ts:211]: We should, but we need to first initialize tests.

File: projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts (hunk 2 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts]: Ah, my fault, Michael change breakpoints to be always of type number. The others are now handled by the color-attribute-input.

795bc324f66ff98dc00435d0a506a018f6e340ea

File: projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts (hunk 1 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/colors/color-breakpoint-input/color-breakpoint-input.component.ts]: You are right, I fixed it.
795bc324f66ff98dc00435d0a506a018f6e340ea

**Pull request merged**